src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/AbstractConverter.cs

```diff
@@ -183,6 +183,15 @@ private static QueryClauseSyntax CreateQueryClause(ExtendedSyntaxNode node)
             //      OR
             //   c1.SelectMany(n1 => ...
             //
+
+            // Avoid `.Select(x => x)`
+            if (lambdaBody is IdentifierNameSyntax identifier &&
+                identifier.Identifier.ValueText == forEachStatement.Identifier.ValueText &&
+                receiverForInvocation is InvocationExpressionSyntax receiverInvocationExpression)
+            {
+                return receiverInvocationExpression;
+            }
```

Comment:
shouldn't you only do this if `!hasForEachChild`?  Otherwise, this will elide `.SelectMany(x => x)` right? (can we also add a test for this)?

Reply:
Thanks. Done with code change (but no tests yet).

Reply:
@CyrusNajmabadi Does this look like a good enough example, or were you thinking of something else?

![image](https://user-images.githubusercontent.com/235241/68414770-a755a300-0145-11ea-91e8-1bf1ed596290.png)

Reply:
Added the above as a test: QueryWhichRequiresSelectManyWithIdentityLambda

Reply:
That's perfect :)